src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -337,7 +337,10 @@ private ImmutableArray<ResourceMetadata> GetAllResourceMetadata()
                 {
                     foreach (var output in model.Outputs)
                     {
-                        this.ResourceMetadata.TryAdd(moduleSymbol, output.Name);
+                        if (this.ResourceMetadata.TryAdd(moduleSymbol, output.Name) is { } resource)
+                        {
+                            resources.Add(resource);
```

Comment:
I'd check with @rynowak on this one - there may have been a deliberate reason not to consider module output resources to be part of the `AllResources` collection

Reply:
Backed this out per your suggestion for EmitLimitationCalculator, but I would still like to hear from @rynowak on this one. Just based on the name of `AllResources`, this looked like an error to me.